### PR TITLE
modernize

### PR DESCRIPTION
**Pull request by achille-roussel**
This PR modernizes the package implementation:
- use classes instead of hacking prototypes
- move tests to ava
- use an actual UDP server in tests instead of mocking private methods
- add tests for gauges and histograms
- removed all dependencies

### CODE REVIEW

**Comment by achille-roussel** [comment]
Thanks for the review!

File: package.json [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
     "statsd"
   ],
-  "dependencies": {
-    "debug": "^2.6.9",
-    "forward-events": "0.0.1"
-  },
+  "dependencies": {},
   "devDependencies": {
-    "mocha": "*"
+    "ava": "^3.3.0",
```
Comment (mericsson) [inline package.json]: ava is okay.. but I think generally segment is coalescing around jest -- see example template: https://github.com/segmentio/ts-node-service-template/

Comment (mericsson) [reply package.json]: probably would've been an easier conversion from mocha --> jest too.

Comment (mericsson) [reply package.json]: probably time to 'publish' this doc and discuss internally for our team https://paper.dropbox.com/doc/Integrations-JS-Code-Style--Auco1OGKypMy9atBob16ZM~eAg-lQyB0y2Ed7X3b5LgIAY14

Comment (achille-roussel) [reply package.json]: 👍 

Never gets old https://dayssincelastjavascriptframework.com/

Comment (mericsson) [reply package.json]: :) for sure -- at least jest is part of the facebook ecosystem which seems to be getting 'standardize' treatment? eh..

**Comment by mericsson** [comment]
@achille-roussel forgot to ask - what prompted this change? node 12 upgrade?

**Comment by achille-roussel** [comment]
I was taking a dependency on this package and realized it needed a bit of cleanup

**Pull request merged**